src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_x509.c

```diff
@@ -52,7 +53,27 @@ AppleCryptoNative_X509GetPublicKey(SecCertificateRef cert, SecKeyRef* pPublicKey
     if (cert == NULL || pPublicKeyOut == NULL || pOSStatusOut == NULL)
         return kErrorBadInput;
 
-    *pOSStatusOut = SecCertificateCopyPublicKey(cert, pPublicKeyOut);
+    // SecCertificateCopyPublicKey was deprecated in 10.14, so use SecCertificateCopyKey on the systems that have it (10.14+),
+    // and SecCertificateCopyPublicKey on the systems that don’t.
+    static SecKeyRef (*secCertificateCopyKey)(SecCertificateRef);
+    static OSStatus (*secCertificateCopyPublicKey)(SecCertificateRef, SecKeyRef*);
+    static int checked;
+    
+    if (!checked)
+    {
+        secCertificateCopyKey = (SecKeyRef (*)(SecCertificateRef))dlsym(RTLD_DEFAULT, "SecCertificateCopyKey");
+        secCertificateCopyPublicKey = (OSStatus (*)(SecCertificateRef, SecKeyRef*))dlsym(RTLD_DEFAULT, "SecCertificateCopyPublicKey");
+        checked = 1;
```

Comment:
In C# I'd do this as `Volatile.Write(ref checked, 1)`, or declare `checked` as `volatile`.  Is declaring `checked` `volatile` here the right way to prevent a reordering race condition, @janvorli?

Reply:
(Or is my `static local`-fu weak, and we don't need to do a lazy initializer manually, because static locals guarantee that they only run their initializer once? (and therefore the two dlsym calls should move up to variable initializers))